timesketch/lib/analyzers/feature_plugins/winevt.py

```diff
@@ -0,0 +1,172 @@
+                feature extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction for `feature_name`.
+        """
+
+        if not name:
+            raise ValueError("Feature name is empty")
+
+        if not config:
+            raise ValueError("Feature configuration value is empty")
+
+        source_name = config.get("source_name", None)
+        if not source_name:
+            raise ValueError("source_name is empyt")
+
+        if not isinstance(source_name, list):
+            raise TypeError(
+                f"Expecting source_name as a list in {name} and got {type(source_name)}"
+            )
+
+        source_name = source_name[0]
+
+        # Using -1 as a default value for event_identifier and event_version as
+        # 0 is considered as None.
+        event_identifier = int(config.get("event_identifier", -1))
+        event_version = int(config.get("event_version", -1))
+
+        if not source_name:
+            raise ValueError("source_name is empty")
+
+        if event_identifier < 0:
+            raise ValueError(
+                f"event_identifier is {event_identifier}. Expecting the value >= 0"
+            )
+
+        if event_version < 0:
+            raise ValueError(f"event_version is {event_version}. Expecting value >= 0")
+
+        mappings = config.get("mapping", None)
+        if not mappings:
+            raise ValueError(f"mapping value for {name} is empty")
+
+        query = (
+            f"source_name: {source_name} AND event_identifier: {event_identifier}"
+            f" AND event_version: {event_version}"
+        )
+
+        events = self.analyzer_object.event_stream(
+            query_string=query, return_fields=self.EVENT_FIELDS
+        )
+        event_counter = 0
+
+        for event in events:
+            attributes = {}
+
+            strings = event.source.get("strings", None)
+            if not strings:
+                logger.debug("No strings attribute in event ID: %s", event.event_id)
+                continue
+
+            for mapping in mappings:
+                attribute_name = mapping.get("name", None)
+                string_index = int(mapping.get("string_index", -1))
+                attribute_aliases = mapping.get("aliases", None)
+
+                attribute_value = ""
+                try:
+                    attribute_value = strings[string_index]
+                except IndexError:
+                    logger.warning(
+                        "The index %d does not exist in strings", string_index
+                    )
+                    continue
+
+                attributes[attribute_name] = attribute_value
+                if attribute_aliases:
+                    for alias in attribute_aliases:
+                        attributes[alias] = attribute_value
+
+            event.add_attributes(attributes)
+            event.commit()
+
+            event_counter += 1
+
+        return f"{event_counter} events updated by {name}."
```

Comment:
```suggestion
        return f"winevt feature extraction [{name}] extracted {event_counter} features."
```

For now let's keep the return value similar to the feature extraction one. Later we can update them to the new verbose output as well.

Reply:
Done.